sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateKitDAO.java

```diff
@@ -0,0 +1,192 @@
+
+  protected static final Logger log = LoggerFactory.getLogger(HibernateKitDAO.class);
+
+  @Autowired
+  private SessionFactory sessionFactory;
+  @Autowired
+  private JdbcTemplate jdbcTemplate;
+
+  public JdbcTemplate getJdbcTemplate() {
+    return jdbcTemplate;
+  }
+
+  public void setJdbcTemplate(JdbcTemplate jdbcTemplate) {
+    this.jdbcTemplate = jdbcTemplate;
+  }
+
+  private Session currentSession() {
+    return getSessionFactory().getCurrentSession();
+  }
+
+  public SessionFactory getSessionFactory() {
+    return sessionFactory;
+  }
+
+  public void setSessionFactory(SessionFactory sessionFactory) {
+    this.sessionFactory = sessionFactory;
+  }
+
+  @Override
+  public Kit get(long id) throws IOException {
+    return (Kit) currentSession().get(KitImpl.class, id);
+  }
+
+  @CoverageIgnore
+  @Override
+  public Kit lazyGet(long id) throws IOException {
+    return get(id);
+  }
+
+  @Override
+  public Kit getKitByIdentificationBarcode(String barcode) throws IOException {
+    Criteria criteria = currentSession().createCriteria(KitImpl.class);
+    criteria.add(Restrictions.eq("identificationBarcode", barcode));
+    return (Kit) criteria.uniqueResult();
+  }
+
+  @Override
+  public Kit getKitByLotNumber(String lotNumber) throws IOException {
+    Criteria criteria = currentSession().createCriteria(KitImpl.class);
+    criteria.add(Restrictions.eq("lotNumber", lotNumber));
+    return (Kit) criteria.uniqueResult();
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public Collection<Kit> listAll() throws IOException {
+    return currentSession().createCriteria(KitImpl.class).list();
+  }
+
+  @Override
+  public int count() throws IOException {
+    long c = (Long) currentSession().createCriteria(KitImpl.class).setProjection(Projections.rowCount()).uniqueResult();
+    return (int) c;
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public List<Kit> listKitsByType(KitType kitType) throws IOException {
+    Criteria criteria = currentSession().createCriteria(KitImpl.class);
+    criteria.createAlias("kitDescriptor", "kd");
+    criteria.add(Restrictions.eq("kd.kitType", kitType));
+    return criteria.list();
+  }
+
+  @Override
+  public long save(Kit kit) throws IOException {
+    long id;
+    if (kit.getId() == KitImpl.UNSAVED_ID) {
+      id = (Long) currentSession().save(kit);
+    } else {
+      currentSession().update(kit);
+      id = kit.getId();
+    }
+    return id;
+  }
+
+  @Override
+  public KitDescriptor getKitDescriptorById(long id) throws IOException {
+    return (KitDescriptor) currentSession().get(KitDescriptor.class, id);
+  }
+
+  @Override
+  public KitDescriptor getKitDescriptorByPartNumber(String partNumber) throws IOException {
+    Criteria criteria = currentSession().createCriteria(KitDescriptor.class);
+    criteria.add(Restrictions.eq("partNumber", partNumber));
+    return (KitDescriptor) criteria.uniqueResult();
```

Comment:
There is no unique key on this in the db, and we have a number of items with partNumber `1` so I don't know that this will behave as desired.